ide-backend/IdeSession/Update.hs

```diff
@@ -179,15 +179,21 @@ initSession initParams@SessionInitParams{..} ideConfig@SessionConfig{..} = do
   verifyConfig ideConfig
 
   configDirCanon <- Dir.canonicalizePath configDir
-  ideSessionDir  <- createTempDirectory configDirCanon "session."
-  let ideStaticInfo = IdeStaticInfo{..}
+  ideSessionDir' <- createTempDirectory configDirCanon "session."
+  ideSessionDir <- case configLocalWorkingDir of
```

Comment:
ideSessionDir is being set to the local working directory so that ide-backend doesn't start looking for data files that the source files depend on in the session.nnnn directory.